lblod/frontend-gelinkt-notuleren-publicatie
# Enable embroider

## Overview
This PR enables embroider as the build system for this application. 
Specifically, it adds the following packages:
- `@embroider/compat`
- `@embroider/core`
- `@embroider/router`
- `@embroider/webpack`

It replaces the `app.toTree` call by `require('@embroider/compat').compatBuild`.

For now the following embroider options are enabled:
- `staticAddonTestSupportTrees`
- `staticAddonTrees`
- `staticHelpers`
- `staticModifiers`
- `staticComponents`
- lazy route splitting is not yet enabled


### How to test/reproduce
- start the application as usual
- observe the faster build times and hot reload and check if the application works as before

### Challenges/uncertainties
On `fastboot`, `embroider` and `yasgui`:
the `yasgui` package requires the browser API in order to work => this causes the library not to work in fastboot mode. This is why we run yasgui in the browser and not in fastboot mode, as follows:
```hbs
{{#if this.fastboot.isFastBoot}}
  <AuHeading @level="2" @skin="4">Aan het laden </AuHeading>
  <AuLoader />
{{else}}
  <div {{yasgui}}></div>
{{/if}}
```

This ensures yasgui has complete access to the browser API.
When enabling the `staticHelpers`, `staticModifiers` and `staticComponents` (these need to be enabled all at once), an error shows up indicating yasgui cannot access browser API's. However this does not block the app from running and the `/sparql` page (which uses yasgui) works as expected.

### Checks PR readiness
